routing/matcher.go

```diff
@@ -358,8 +376,10 @@ func (m *matcher) match(r *http.Request) (*Route, map[string]string) {
 		path = path[:len(path)-1]
 	}
 
+	lrm := &leafRequestMatcher{r, path}
+
 	// first match fixed and wildcard paths
-	leaves, params := matchPathTree(m.paths, path)
+	leaves, params := matchPathTree(m.paths, path, lrm)
 	l := matchLeaves(leaves, r, path)
```

Comment:
perhaps this double work, since in lrm already this work done. Just need somehow get leaf from lrm and this can save some performance.